src/wasm/wasm-s-parser.cpp

```diff
@@ -1819,20 +1819,6 @@ void SExpressionWasmBuilder::parseExport(Element& s) {
     } else {
       throw ParseException("invalid export");
     }
-  } else if (!s[2]->dollared() && !std::isdigit(s[2]->str()[0])) {
-    ex->value = s[3]->str();
-    if (s[2]->str() == MEMORY) {
-      if (!wasm.memory.exists) {
-        throw ParseException("memory exported but no memory");
-      }
-      ex->kind = ExternalKind::Memory;
-    } else if (s[2]->str() == TABLE) {
-      ex->kind = ExternalKind::Table;
-    } else if (s[2]->str() == GLOBAL) {
-      ex->kind = ExternalKind::Global;
-    } else {
-      throw ParseException("invalid ext export");
-    }
```

Comment:
This is in `parseExport`. For exports, the syntax I know is
```
(export "base" (global|func|... $name))
(export "base" (global|func|... index)) // index is a number
```
And I don't know which syntax corresponds to this code. This requires `s[2]` to be a string, but none of the formats above meets that requirement.

This logic was first added in https://github.com/WebAssembly/binaryen/commit/a0156dbc900c642dd9f1b8f44ab13c06ff2e8fb9 (memory only) and https://github.com/WebAssembly/binaryen/commit/2e2e0241a10c0eccb89365c1b50217d6d0d3a386 (other types), but the only test case relevant that was added with these two commits was
 ```
(export "memory" memory)
```

Do we still support this syntax? This line was also deleted in https://github.com/WebAssembly/binaryen/commit/42ad2cdbe5baa32dd1b0aea143a4e45f81e8b9b5 and never appeared ever since, and this is the only line I have seen that triggers this part of the code.